docs/source/WritingTests/WritingSimpleTests.rst

```diff
@@ -167,6 +162,42 @@ to pick up a living guest, connect to it via ssh, and return its uptime.
            logging.info("Guest uptime result is: %s", uptime)
            session.close()
 
+#. Note that all failures that might happen here are implicitly handled
+   by the methods called. If a test went from its beginning to its end
+   without unhandled exceptions, avocado assumes the test automatically
+   as **PASS**, *no need to mark a test as explicitly passed*. If you
+   have explicit points of failure, for more complex tests, you might
+   want to mark it explicitly. ``test.skip`` make the test **SKIP**,
+   ``test.error`` make the test **Error**, and ``test.fail`` make the
+   test **Fail**. Note, people should never import ``exceptions``
+   from ``avocado.core`` to raise them in test case, see `avocado doc <http://avocado-framework.readthedocs.io/en/latest/api/core/avocado.core.html>`_
+   for more details. *BTW, check the uptime makes no sense, but let's continue
+   this example for test status explanation*:
+
+   .. code-block:: python
+      :emphasize-lines: 16,18,20
+
+       def run(test, params, env):
+           """
+           Docstring describing uptime.
+           """
+           vm = env.get_vm(params["main_vm"])
+           vm.verify_alive()
+           timeout = float(params.get("login_timeout", 240))
+           session = vm.wait_for_login(timeout=timeout)
+           uptime = session.cmd('uptime')
+           logging.info("Guest uptime result is: %s", uptime)
+           session.close()
+           expected_skip_msg = '0 min'
+           expected_err_msg = '1 day'
+           expected_fail_msg = '10 days'
+           if expected_skip_msg in uptime:
+               test.skip('Skip message')
```

Comment:
yes as @ldoktor mentioned, test.skip() gives error msg suggesting to use test.cancel() from inside test,
```
2017-04-05 02:26:41,069 stacktrace       L0044 ERROR| Traceback (most recent call last):
2017-04-05 02:26:41,070 stacktrace       L0044 ERROR|   File "/usr/local/lib/python2.7/dist-packages/avocado_plugins_vt-48.0-py2.7.egg/avocado_vt/test.py", line 260, in runTest
2017-04-05 02:26:41,070 stacktrace       L0044 ERROR|     raise self.__status
2017-04-05 02:26:41,070 stacktrace       L0044 ERROR| TestSetupSkip: [WARNING: self.skip() will be deprecated. Use 'self.cancel()' or the skip decorators] Test Skip
2017-04-05 02:26:41,070 stacktrace       L0045 ERROR|
2017-04-05 02:26:41,070 test             L0748 ERROR| ERROR 1-migration.qemu.qcow2.virtio_scsi.smp2.virtio_net.Ubuntu.17.04.ppc64le.powerkvm-libvirt.virsh.exception.skip -> TestError: Calling skip() in places other than setUp() is not allowed in avocado, you must fix your test. Original skip exception: [WARNING: self.skip() will be deprecated. Use 'self.cancel()' or the skip decorators] Test Skip
```